internal/digraph/executor/docker.go

```diff
@@ -108,28 +111,65 @@ func (e *docker) Run(ctx context.Context) error {
 		args = append(args, val)
 	}
 
-	// If containerName is set, use exec instead of creating a new container
-	if e.containerName != "" {
+	if e.image == "" {
 		return e.execInContainer(ctx, cli, args)
 	}
 
-	// New container creation logic
-	if e.pull {
-		reader, err := cli.ImagePull(ctx, e.image, image.PullOptions{})
+	// If platform is not provided, imageReference will be set to the input image name
+	// else, it will be set to the locally available imageId whose platform matches the input platform
+	// or an empty string to signify that we don't have the right image locally.
+	imageReference := ""
+
+	// Only inspect image if pull mode is not always
+	if e.pull != "always" {
```

Comment:
Could we extract this into a separate method? It’s starting to get a bit complicated.